openpgp/packet/signature.go

```diff
@@ -304,12 +304,14 @@ func parseSignatureSubpacket(sig *Signature, subpacket []byte, isHashed bool) (r
 	isCritical = subpacket[0]&0x80 == 0x80
 	subpacket = subpacket[1:]
 	sig.rawSubpackets = append(sig.rawSubpackets, outputSubpacket{isHashed, packetType, isCritical, subpacket})
+	if !isHashed &&
+		packetType != issuerSubpacket &&
+		packetType != issuerFingerprintSubpacket &&
+		packetType != embeddedSignatureSubpacket {
+		return
+	}
 	switch packetType {
 	case creationTimeSubpacket:
-		if !isHashed {
```

Comment:
As you had mentioned, we still need to enforce that the hashed area has a creationTimeSubpacket, which is different from what we checked so far.

The RFC says

> MUST be present in the hashed area.

Is that enforced somewhere ? I could not find such check

Reply:
Yeah, it's enforced here: https://github.com/ProtonMail/go-crypto/blob/2fdefb57b69d1143b9f3d55bfbdd7953f3bb111a/openpgp/packet/signature.go#L235-L237, in the first call to `parseSignatureSubpackets` (which is for the hashed subpackets). It could be made more explicit but the check is there.

Reply:
Allright thanks